src/utils/daos_hdlr.c

```diff
@@ -1645,6 +1656,86 @@ fs_copy_file(struct cmd_args_s *ap,
 	return rc;
 }
 
+static int
+fs_copy_symlink(struct cmd_args_s *ap,
+		struct file_dfs *src_file_dfs,
+		struct file_dfs *dst_file_dfs,
+		struct stat *src_stat,
+		const char *src_path,
+		const char *dst_path)
+{
+	int		rc = 0;
+	daos_size_t	len = 0;
+	char		*symlink_value = NULL;
+
+	D_ALLOC(symlink_value, DFS_MAX_PATH);
+	if (symlink_value == NULL)
+		D_GOTO(out_copy_symlink, rc = -DER_NOMEM);
+
+	len = PATH_MAX - 1;
```

Comment:
You're using both DFS_MAX_PATH and PATH_MAX here, perhaps do the following for safety.
```suggestion
	daos_size_t	len = DFS_MAX_PATH;
	char		*symlink_value;

	D_ALLOC(symlink_value, len + 1);
	if (symlink_value == NULL)
		D_GOTO(out_copy_symlink, rc = -DER_NOMEM);

```